misc/connection-limits/src/lib.rs

```diff
@@ -327,11 +325,43 @@ impl NetworkBehaviour for Behaviour {
                     .entry(peer_id)
                     .or_default()
                     .remove(&connection_id);
+
+                self.pending_inbound_connections.remove(&connection_id);
+                self.pending_outbound_connections.remove(&connection_id);
```

Comment:
@AgeManning are these two lines needed? A `FromSwarm::ConnectionClosed` should always be preceded by a `handle_established_{outbound,inbound}_connection`. In those methods we reduce the `self.pending_*_connections` counters.

Reply:
Removed with 8167c711e. Please still review @AgeManning.

Reply:
I agree this is fine to remove. Just being overly cautious